File: src/main/java/com/vielheit/graph/repository/EntryRepository.java
package com.vielheit.graph.repository;

import com.vielheit.graph.domain.Entry;
import org.springframework.data.neo4j.annotation.Query;
import org.springframework.data.neo4j.repository.GraphRepository;
import org.springframework.data.repository.query.Param;
import org.springframework.stereotype.Repository;

import java.util.List;

@Repository
public interface EntryRepository extends GraphRepository<Entry> {

    @Query("MATCH (user:GraphUser {userId: {userId}})-[:OWNS]->(entries:Entry) RETURN entries")
    List<Entry> findByUserId(@Param("userId") Long userId);
}

File: src/main/kotlin/cc/kojeve/vielheit/repository/EntryRepository.kt
package cc.kojeve.vielheit.repository

import cc.kojeve.vielheit.domain.Entry
import cc.kojeve.vielheit.domain.Tag
import cc.kojeve.vielheit.domain.User
import org.springframework.data.repository.CrudRepository
import javax.transaction.Transactional

interface EntryRepository : CrudRepository<Entry, Long> {
    fun findByUser(user: User): Set<Entry>

    fun findByUserAndTags(user: User, tag: Tag): Set<Entry>

}